docs/using-the-jdbc-driver/using-plugins/UsingTheDeveloperPlugin.md

```diff
@@ -0,0 +1,43 @@
+## Developer Plugin
+
+The Developer Plugin allows developers to inject a prepared exception to a connection, let the connection to raise this exception on the next call and let an application to process it. Some of the exception that the driver may raise is quite hard to get and it's by the nature of such exceptions. For example, network issues may cause the driver to raise various timeout exceptions. However it may require substantial efforts to design and to build a testing environment where such timeout exceptions could be produced with 100% accuracy and 100% guarantee. If test suite can't verify such network outages cases with 100% accuracy it significantly decreases the value of such tests and makes the tests flaky. The Developer Plugins simplifies testing of such scenarios as shown below.
+
+> :warning: The plugin is NOT intended to be used in production environments. It's designed for the purpose of testing.
+
+### Simulate an exception while opening a new connection
+
+In order to raise a test exception while opening a new connection, it's possible to use global `ExceptionSimulatorManager`. Once the exception is raised it's cleared up in `ExceptionSimulatorManager` and another attempt to open a new connection goes normal.
+
+```
+final Properties properties = new Properties();
+properties.setProperty(PropertyDefinition.PLUGINS.name, "dev");
+
+final SQLException testExceptionToRaise = new SQLException("test");
+ExceptionSimulatorManager.raiseExceptionOnNextConnect(testExceptionToRaise);
+
+Connection connection = DriverManager.getConnection("jdbc:aws-wrapper:postgresql://db-host/", properties); // that throws the exception
+
+Connection anotherConnection = DriverManager.getConnection("jdbc:aws-wrapper:postgresql://db-host/", properties); // it goes normal with no exception
+```
+
+### Simulate an exception with already opened connection
+
+A new `dev` plugin should be added to connection plugins parameter in order to be able to intercept JDBC calls and raise a test exception when conditions are met. Similar to previous case, the exception is cleared up once it's raised and subsequent JDBC calls should go normal.  
```

Comment:
1- i think we should mention to add `dev` to properties before we get here, once at the top and that's it

2 - Mention that the plugin needs to be in the connection properties in order to use it with an open connection. ex:
```
It is possible to also simulate an exception thrown in a connection after the connection has been opened. It is only possible if the `dev` plugin has been added to the connection properties beforehand.
```

3- The first example uses `raiseExceptionOnConnect` and this one uses `raiseExceptionOnNextCall` with parameters.
Maybe somewhere in the doc we should list all the possibilities/calls of the plugin.